vector/src/test/java/im/vector/app/features/settings/devices/v2/othersessions/OtherSessionsViewModelTest.kt

```diff
@@ -0,0 +1,292 @@
+        val viewModelTest = viewModel.test()
+        viewModel.handle(OtherSessionsAction.ToggleSelectionForDevice(A_DEVICE_ID))
+
+        // Then
+        viewModelTest
+                .assertLatestState { state -> state == expectedState }
+                .finish()
+    }
+
+    @Test
+    fun `given select all action when handling the action then viewState is updated with correct info`() {
+        // Given
+        val deviceFullInfo1 = givenDeviceFullInfo(A_DEVICE_ID, isSelected = false)
+        val deviceFullInfo2 = givenDeviceFullInfo(A_DEVICE_ID, isSelected = true)
+        val devices: List<DeviceFullInfo> = listOf(deviceFullInfo1, deviceFullInfo2)
+        givenGetDeviceFullInfoListReturns(filterType = defaultArgs.defaultFilter, devices)
+        val expectedState = OtherSessionsViewState(
+                devices = Success(listOf(deviceFullInfo1.copy(isSelected = true), deviceFullInfo2.copy(isSelected = true))),
+                currentFilter = defaultArgs.defaultFilter,
+                excludeCurrentDevice = defaultArgs.excludeCurrentDevice,
+                isSelectModeEnabled = false,
+        )
+
+        // When
+        val viewModel = createViewModel()
+        val viewModelTest = viewModel.test()
+        viewModel.handle(OtherSessionsAction.SelectAll)
+
+        // Then
+        viewModelTest
+                .assertLatestState { state -> state == expectedState }
+                .finish()
+    }
+
+    @Test
+    fun `given deselect all action when handling the action then viewState is updated with correct info`() {
+        // Given
+        val deviceFullInfo1 = givenDeviceFullInfo(A_DEVICE_ID, isSelected = false)
+        val deviceFullInfo2 = givenDeviceFullInfo(A_DEVICE_ID, isSelected = true)
+        val devices: List<DeviceFullInfo> = listOf(deviceFullInfo1, deviceFullInfo2)
+        givenGetDeviceFullInfoListReturns(filterType = defaultArgs.defaultFilter, devices)
+        val expectedState = OtherSessionsViewState(
+                devices = Success(listOf(deviceFullInfo1.copy(isSelected = false), deviceFullInfo2.copy(isSelected = false))),
+                currentFilter = defaultArgs.defaultFilter,
+                excludeCurrentDevice = defaultArgs.excludeCurrentDevice,
+                isSelectModeEnabled = false,
+        )
+
+        // When
+        val viewModel = createViewModel()
+        val viewModelTest = viewModel.test()
+        viewModel.handle(OtherSessionsAction.DeselectAll)
+
+        // Then
+        viewModelTest
+                .assertLatestState { state -> state == expectedState }
+                .finish()
+    }
+
+    private fun givenGetDeviceFullInfoListReturns(
+            filterType: DeviceManagerFilterType,
+            devices: List<DeviceFullInfo>,
+    ) {
+        every { fakeGetDeviceFullInfoListUseCase.execute(filterType, any()) } returns flowOf(devices)
+    }
+
+    private fun givenDeviceFullInfo(deviceId: String, isSelected: Boolean): DeviceFullInfo {
```

Comment:
Maybe rename to
```suggestion
    private fun aDeviceFullInfo(deviceId: String, isSelected: Boolean): DeviceFullInfo {
```
And move to a Fixture object?

Reply:
Yes fixed here: [431bd8](https://github.com/vector-im/element-android/pull/7408/commits/d431bd814aa909a793edab04261d6c16386ab206)